satellite/satellitedb/overlaycache.go

```diff
@@ -22,6 +24,251 @@ type overlaycache struct {
+		overlayNode := &dbx.OverlayCacheNode{}
+		err = rows.Scan(&overlayNode.NodeId, &overlayNode.NodeType,
+			&overlayNode.Address, &overlayNode.FreeBandwidth, &overlayNode.FreeDisk,
+			&overlayNode.AuditSuccessRatio, &overlayNode.AuditUptimeRatio,
+			&overlayNode.AuditCount, &overlayNode.AuditSuccessCount,
+			&overlayNode.UptimeCount, &overlayNode.UptimeSuccessCount)
+		if err != nil {
+			return nil, err
+		}
+
+		node, err := convertOverlayNode(overlayNode)
+		if err != nil {
+			return nil, err
+		}
+		reputableNodes = append(reputableNodes, node)
+	}
+
+	return reputableNodes, nil
+}
+
+func (cache *overlaycache) filterNewNodes(ctx context.Context, req *filterNewRequest) (newNodes []*pb.Node, err error) {
+	rows, err := cache.findNewNodesQuery(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = utils.CombineErrors(err, rows.Close())
+	}()
+
+	for rows.Next() {
+		overlayNode := &dbx.OverlayCacheNode{}
+		err = rows.Scan(&overlayNode.NodeId, &overlayNode.NodeType,
+			&overlayNode.Address, &overlayNode.FreeBandwidth, &overlayNode.FreeDisk,
+			&overlayNode.AuditSuccessRatio, &overlayNode.AuditUptimeRatio,
+			&overlayNode.AuditCount, &overlayNode.AuditSuccessCount,
+			&overlayNode.UptimeCount, &overlayNode.UptimeSuccessCount)
+		if err != nil {
+			return nil, err
+		}
+
+		node, err := convertOverlayNode(overlayNode)
+		if err != nil {
+			return nil, err
+		}
+		newNodes = append(newNodes, node)
+	}
+
+	return newNodes, nil
+}
+
+func (cache *overlaycache) findReputableNodesQuery(ctx context.Context, req *filterReputableRequest) (*sql.Rows, error) {
+	auditCount := req.minReputation.AuditCount
+	auditSuccessRatio := req.minReputation.AuditSuccessRatio
+	uptimeCount := req.minReputation.UptimeCount
+	uptimeRatio := req.minReputation.UptimeRatio
+	nodeAmt := req.reputableNodeAmount
+
+	var args []interface{}
+	var rows *sql.Rows
+	var err error
+	var nodeTypeStorage int32 = 2
+
+	if len(req.excluded) == 0 {
+		args = append(args, auditCount, req.newNodeAuditThreshold,
+			auditSuccessRatio, uptimeCount, uptimeRatio,
+			req.freeBandwidth, req.freeDisk, nodeTypeStorage, nodeAmt)
+
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT overlay_cache_nodes.node_id,
+	overlay_cache_nodes.node_type, overlay_cache_nodes.address, overlay_cache_nodes.free_bandwidth,
+	overlay_cache_nodes.free_disk, overlay_cache_nodes.audit_success_ratio,
+	overlay_cache_nodes.audit_uptime_ratio, overlay_cache_nodes.audit_count,
+	overlay_cache_nodes.audit_success_count, overlay_cache_nodes.uptime_count,
+	overlay_cache_nodes.uptime_success_count
+	FROM overlay_cache_nodes
+	WHERE overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_count > ?
+	AND overlay_cache_nodes.audit_success_ratio > ?
+	AND overlay_cache_nodes.uptime_count > ?
+	AND overlay_cache_nodes.audit_uptime_ratio > ?
+	AND overlay_cache_nodes.free_bandwidth > ?
+	AND overlay_cache_nodes.free_disk > ?
+	AND overlay_cache_nodes.node_type == ?
+	LIMIT ?
```

Comment:
This wouldn't affect performance much, but you could increase readability some by omitting the table name from everything (because it's the default while querying from one table). Something like

```
SELECT node_id, node_type, address, free_bandwidth, free_disk, audit_success_ratio,
	audit_uptime_ratio, audit_count, audit_success_count, uptime_count, uptime_success_count
FROM overlay_cache_nodes
WHERE audit_count > ?
	AND audit_count > ?
	AND audit_success_ratio > ?
	AND uptime_count > ?
	AND audit_uptime_ratio > ?
	AND free_bandwidth > ?
	AND free_disk > ?
	AND node_type = ?
LIMIT ?
```

Reply:
oh awesome, I didn't know about this thanks!